app/predictor.py

```diff
@@ -30,10 +30,28 @@ def __call__(self, file: IO) -> np.ndarray:
         image = np.expand_dims(image, axis=0)
         return self.model.predict(image)
 
-    @staticmethod
-    def _load_image_as_array(file: IO) -> np.ndarray:
+    @property
+    def image_shape(self) -> Tuple[int, int]:
+        return self.model.input_shape[1:]
+
+    def _load_image_as_array(self, file: IO) -> np.ndarray:
+        """Loads an image from file and applies pre-processing steps to conform to the model input
+
+        Args:
+            file (IO): A file-like object pointing to the image file
+
+        Returns:
+            np.ndarray: The image as an array that is compatible with the model
+        """
         with Image.open(file) as image:
-            return np.asarray(image)
+            if self._model_is_grayscale():
```

Comment:
Not covering the case where the model can handle `RGB` images and the input is grayscale